packages/react/src/app-layout/AppLayoutSidebarNav.tsx

```diff
@@ -0,0 +1,263 @@
+					color: boxPalette[isActive ? 'foregroundText' : 'foregroundAction'],
+
+					...(isActive && {
+						fontWeight: tokens.fontWeight.bold,
+						background: boxPalette.backgroundShade,
+						color: boxPalette.foregroundText,
+					}),
+
+					...(hasEndElement && {
+						'& > :last-child': {
+							marginLeft: 'auto',
+						},
+					}),
+
+					...focusStyles,
+
+					'&:hover': {
+						background: boxPalette.backgroundShade,
+						color: boxPalette.foregroundText,
+						'& span:first-of-type': {
+							textDecoration: 'underline',
+						},
+					},
+				},
+			}}
+		>
+			{children}
+		</li>
+	);
+}
+
+function AppLayoutSidebarNavDivider() {
+	return (
+		<Box as="li" paddingY={1} paddingX={1.5} aria-hidden="true">
+			<hr
+				css={{
+					boxSizing: 'content-box',
+					height: 0,
+					margin: 0,
+					overflow: 'visible',
+					border: 'none',
+					borderTopWidth: tokens.borderWidth.sm,
+					borderTopStyle: 'solid',
+					borderColor: boxPalette.borderMuted,
+					width: '100%',
+				}}
+			/>
+		</Box>
+	);
+}
+
+function HideMenuButton() {
+	const { hideMenu, hideMenuButtonRef } = useAppLayoutContext();
+	return (
+		<Flex
+			height={HEADER_HEIGHT.md}
+			alignItems="center"
+			borderBottom
+			borderColor="muted"
+			paddingLeft={{ xs: 1, lg: 0 }}
+			flexShrink={0}
+		>
+			<BaseButton
+				ref={hideMenuButtonRef}
+				onClick={hideMenu}
+				css={{
+					// Mobile button styles
+					'> span:first-of-type': {
+						width: '100%',
+						display: 'flex',
+						flexDirection: 'column',
+						alignItems: 'center',
+						gap: mapSpacing(0.5),
+						boxSizing: 'border-box',
+						padding: mapSpacing(1),
+						color: boxPalette.foregroundAction,
+						...fontGrid('xs', 'default'),
+					},
+
+					// Desktop button styles
+					'> span:last-of-type': {
+						display: 'none', // Hide on mobile
+						width: '100%',
+						alignItems: 'center',
+						gap: mapSpacing(1),
+						boxSizing: 'border-box',
+						paddingLeft: mapSpacing(1.5),
+						paddingRight: mapSpacing(1.5),
+						paddingTop: mapSpacing(0.5),
+						paddingBottom: mapSpacing(0.5),
+						color: boxPalette.foregroundAction,
+						...fontGrid('xs', 'default'),
+
+						'&:hover': {
+							background: boxPalette.backgroundShade,
+							color: boxPalette.foregroundText,
+						},
+					},
+
+					...focusStyles,
+
+					[tokens.mediaQuery.min.lg]: {
+						width: '100%',
+						// Hide mobile button
+						'> span:first-of-type': { display: 'none' },
+						// Show desktop button
+						'> span:last-of-type': { display: 'flex' },
+					},
+				}}
+			>
+				{/** Mobile button content */}
+				<span>
+					<CloseIcon />
+					<span aria-hidden="true">Close</span>
+					<VisuallyHidden>Close Menu</VisuallyHidden>
+				</span>
+				{/** Desktop button content */}
+				<span>
+					<ChevronsLeftIcon />
+					<span>Hide menu</span>
+				</span>
+			</BaseButton>
+		</Flex>
+	);
+}
+
+const ChevronsLeftIcon = createIcon(
```

Comment:
Should this be moved to @agds-next/react/icons too?

Reply:
Yep, I've added that in this PR https://github.com/steelthreads/agds-next/pull/1086